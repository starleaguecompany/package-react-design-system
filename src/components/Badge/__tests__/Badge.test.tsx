import * as React from 'react'
import { render, screen } from '@testing-library/react'

import { Badge } from '../../../index'

describe('Badge', () => {
  test('should render correctly with no props', () => {
    render(<Badge />)

    expect(screen.getByTestId('Badge')).toMatchSnapshot()
    expect(screen.getByTestId('Badge')).toHaveClass('container', { exact: true })
  })

  test('should render correctly with attributes', () => {
    render(<Badge id="test-id" className="test-class" />)

    expect(screen.getByTestId('Badge')).toMatchSnapshot()
    expect(screen.getByTestId('Badge')).toHaveAttribute('id', 'test-id')
    expect(screen.getByTestId('Badge')).toHaveClass('test-class')
  })

  test('should render correctly with text', () => {
    render(<Badge>Testing badge</Badge>)

    expect(screen.getByTestId('Badge')).toMatchSnapshot()
    expect(screen.getByTestId('Badge')).toHaveTextContent('Testing badge')
    expect(screen.getByText('Testing badge')).toBeVisible()
    expect(screen.getByText('Testing badge')).toHaveClass('container')
  })

  test('should render correctly with text prop', () => {
    render(<Badge text="Testing badge" />)

    expect(screen.getByTestId('Badge')).toMatchSnapshot()
    expect(screen.getByText('Testing badge')).toBeVisible()
    expect(screen.getByTestId('Badge')).toHaveTextContent('Testing badge')
    expect(screen.getByText('Testing badge')).not.toHaveClass('container')
  })

  test('should render correctly with variant prop', () => {
    render(<Badge variant="primary" text="Testing badge" />)

    expect(screen.getByTestId('Badge')).toMatchSnapshot()
    expect(screen.getByText('Testing badge')).toHaveClass('color-gray-primary')
  })

  test('should render correctly with color prop', () => {
    render(<Badge color="green" text="Testing badge" />)

    expect(screen.getByTestId('Badge')).toMatchSnapshot()
    expect(screen.getByText('Testing badge')).toHaveClass('color-green-secondary')
  })

  test('should render correctly with shape prop', () => {
    render(<Badge shape="circle" text="Testing badge" />)

    expect(screen.getByTestId('Badge')).toMatchSnapshot()
    expect(screen.getByText('Testing badge')).toHaveClass('shape-circle')
  })

  test('should render correctly with placement prop', () => {
    render(
      <Badge text="Testing badge" placement="bottom-end">
        <div>Test child</div>
      </Badge>
    )

    expect(screen.getByTestId('Badge')).toMatchSnapshot()
    expect(screen.getByText('Testing badge')).toHaveClass('floating')
    expect(screen.getByText('Testing badge')).toHaveClass('floating-bottom-end')
  })

  test('should render correctly as dot', () => {
    render(<Badge />)

    expect(screen.getByTestId('Badge')).toMatchSnapshot()
    expect(screen.getByTestId('Badge').firstChild).toHaveClass('shape-dot')
  })
})
