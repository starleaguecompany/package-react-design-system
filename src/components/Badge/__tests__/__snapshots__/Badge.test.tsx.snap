// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render correctly as dot 1`] = `
<span
  class="container"
  data-qa="Badge"
>
  <span
    class="element color-gray-secondary shape-dot"
  />
</span>
`;

exports[`Badge should render correctly with attributes 1`] = `
<span
  class="test-class container"
  data-qa="Badge"
  id="test-id"
>
  <span
    class="element color-gray-secondary shape-dot"
  />
</span>
`;

exports[`Badge should render correctly with color prop 1`] = `
<span
  class="container"
  data-qa="Badge"
>
  <span
    class="element color-green-secondary shape-round"
  >
    Testing badge
  </span>
</span>
`;

exports[`Badge should render correctly with no props 1`] = `
<span
  class="container"
  data-qa="Badge"
>
  <span
    class="element color-gray-secondary shape-dot"
  />
</span>
`;

exports[`Badge should render correctly with placement prop 1`] = `
<span
  class="container"
  data-qa="Badge"
>
  <div>
    Test child
  </div>
  <span
    class="element color-gray-secondary shape-round floating floating-bottom-end"
  >
    Testing badge
  </span>
</span>
`;

exports[`Badge should render correctly with shape prop 1`] = `
<span
  class="container"
  data-qa="Badge"
>
  <span
    class="element color-gray-secondary shape-circle"
  >
    Testing badge
  </span>
</span>
`;

exports[`Badge should render correctly with text 1`] = `
<span
  class="container"
  data-qa="Badge"
>
  Testing badge
  <span
    class="element color-gray-secondary shape-dot floating floating-top-end"
  />
</span>
`;

exports[`Badge should render correctly with text prop 1`] = `
<span
  class="container"
  data-qa="Badge"
>
  <span
    class="element color-gray-secondary shape-round"
  >
    Testing badge
  </span>
</span>
`;

exports[`Badge should render correctly with variant prop 1`] = `
<span
  class="container"
  data-qa="Badge"
>
  <span
    class="element color-gray-primary shape-round"
  >
    Testing badge
  </span>
</span>
`;
